src/test/java/com/thirdcc/webapp/authorization/ManagementTeamSecurityExpressionIT.java

```diff
@@ -0,0 +1,305 @@
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(classes = ClubmanagementApp.class)
+@WithMockUser
+public class ManagementTeamSecurityExpressionIT {
+
+    @Autowired
+    private ManagementTeamSecurityExpression managementTeamSecurityExpression;
+
+    @Autowired
+    private EventCrewRepository eventCrewRepository;
+
+    @Autowired
+    private AdministratorRepository administratorRepository;
+
+    @Autowired
+    private YearSessionRepository yearSessionRepository;
+
+    @Autowired
+    private UserService userService;
+
+    private User currentUser;
+    private Administrator administrator;
+    private EventCrew eventCrew;
+    private YearSession yearSession;
+
+    private static final String DEFAULT_YEAR_SESSION_VALUE = "2021/2022";
+    private static final Long DEFAULT_EVENT_ID = 1L;
+    private static final Long DEFAULT_USER_ID = 1L;
+    private static final AdministratorRole DEFAULT_ADMINISTRATOR_ROLE = AdministratorRole.SECRETARY;
+    private static final AdministratorStatus DEFAULT_ADMINISTRATOR_STATUS = AdministratorStatus.ACTIVE;
+
+    private static final String PREVIOUS_YEAR_SESSION_VALUE = "2020/2021";
+    private static final Long OTHER_EVENT_ID = 2L;
+    private static final EventCrewRole EVENT_HEAD_ROLE = EventCrewRole.HEAD;
+    private static final AdministratorRole CC_HEAD_ROLE = AdministratorRole.CC_HEAD;
+
+
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @BeforeEach
+    public void init() {
+        createYearSession();
+        createAdministrator();
+        createEventCrew();
+    }
+
+    private void createYearSession() {
+        yearSession = new YearSession()
+            .value(DEFAULT_YEAR_SESSION_VALUE);
+    }
+
+    private void createEventCrew() {
+        eventCrew = new EventCrew()
+            .eventId(DEFAULT_EVENT_ID)
+            .userId(DEFAULT_USER_ID);
+    }
+
+    private void createAdministrator() {
+        administrator = new Administrator()
+            .role(DEFAULT_ADMINISTRATOR_ROLE)
+            .status(DEFAULT_ADMINISTRATOR_STATUS)
+            .yearSession(DEFAULT_YEAR_SESSION_VALUE)
+            .userId(DEFAULT_USER_ID);
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        yearSessionRepository.deleteAll();
+        eventCrewRepository.deleteAll();
+        administratorRepository.deleteAll();
+    }
+
+    /**
+     * Test on isCurrentCCHead
+     */
+    @Test
+    public void isCurrentCCHead_ShouldReturnTrue() {
+        currentUser = getLoggedInUser();
+
+        initYearSessionDB();
+
+        administrator.setRole(CC_HEAD_ROLE);
```

Comment:
I actually wonder why you don't want to create a new `Administrator`/`EventCrew` for each test case. 🤔 

Cause when we are running test cases, we are running in parallel. And if every test cases are mutating the same instance, there might be a race condition happen.

Reply:
resolved in 472e7e0c05294e6e85bdb0981f53637023935809